lldb/source/Plugins/OperatingSystem/SwiftTasks/OperatingSystemSwiftTasks.cpp

```diff
@@ -142,24 +154,42 @@ StopInfoSP OperatingSystemSwiftTasks::CreateThreadStopReason(
   return thread->GetStopInfo();
 }
 
-std::optional<uint64_t> OperatingSystemSwiftTasks::FindTaskId(Thread &thread) {
+std::optional<addr_t>
+OperatingSystemSwiftTasks::FindTaskAddress(Thread &thread) {
   llvm::Expected<addr_t> task_addr = GetTaskAddrFromThreadLocalStorage(thread);
   if (!task_addr) {
     LLDB_LOG_ERROR(GetLog(LLDBLog::OS), task_addr.takeError(),
                    "OperatingSystemSwiftTasks: failed to find task address in "
                    "thread local storage: {0}");
-    return {};
+    return std::nullopt;
   }
+  return *task_addr;
+}
 
+std::optional<uint64_t>
+OperatingSystemSwiftTasks::FindTaskId(addr_t task_addr) {
   Status error;
   // The Task ID is at offset m_job_id_offset from the Task pointer.
   constexpr uint32_t num_bytes_task_id = 4;
   auto task_id = m_process->ReadUnsignedIntegerFromMemory(
-      *task_addr + m_job_id_offset, num_bytes_task_id, LLDB_INVALID_ADDRESS,
+      task_addr + m_job_id_offset, num_bytes_task_id, LLDB_INVALID_ADDRESS,
       error);
   if (error.Fail())
     return {};
   return task_id;
 }
 
+std::optional<std::string>
+OperatingSystemSwiftTasks::FindTaskName(addr_t task_addr, Process *process) {
+  auto task_name_or_err = GetTaskName(task_addr, process);
+  if (auto err = task_name_or_err.takeError()) {
+    LLDB_LOG_ERROR(GetLog(LLDBLog::OS), std::move(err),
```

Comment:
instead of returning optional and logging the error, you can return Expected and `return llvm::createStringError(...)`. And let the call site do logging or error reporting.

Reply:
or actually, here just return task_name_or_err

Reply:
ah I guess the entire function becomes pointless then :-)